### PR TITLE
Scraper for Tanzania Government Gazettes

### PR DESCRIPTION
**Pull request by ebrahim-j**
**What does this PR do?**

- Adds two spiders for two _different_ domains
- Adds an Item Pipeline for download and storage of file
- Stores the gazettes in a specified directory using the specified structure eg. Year > Month
- Directs the feed exports to an online s3 bucket

**How can this be manually tested**

There is a job on scrapinghub to scrape the TZ gazettes

imageedit_5_7953122664

- step 1: Choose the correct spider to run (`tz_gazettes_v2` for now)
- step 2: Add the year of gazettes wanted to be scraped as an `Argument`
- step 3: Run the job

imageedit_16_3019657827

Check its s3 bucket to find the scrapped gazettes
@DavidLemayian

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Set ENV variables
+    - `SCRAPY_AWS_ACCESS_KEY_ID` - Get this from AWS
+    - `SCARPY_AWS_SECRET_ACCESS_KEY` - Get this from AWS
+    - `SCRAPY_FEED_URI=s3://name-of-bucket-here/gazettes/gazete_index.jsonlines` - Where you want the `jsonlines` output for crawls to be saved. This can also be a local location
+    - `SCRAPY_FILES_STORE=s3://name-of-bucket-here/gazettes` - Where you want scraped gazettes to be stored. This can also be a local location
+
+## Running it Locally
+- To run the spider locally, you can choose to store the scraped files locally to do this set the  ENV variable
+- `SCRAPY_FILES_STORE=/directory/to/store/the/files` which should point to a local folder
+- Then run the command  `scrapy crawl tz_gazeyyes_v2 -a year=2016 -o sl_gazettes.jsonlines`
```
Comment (DavidLemayian) [inline README.md]: Might be a spelling error here. Also, should it be `tz_gazettes.jsonlines`?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - `SCARPY_AWS_SECRET_ACCESS_KEY` - Get this from AWS
+    - `SCRAPY_FEED_URI=s3://name-of-bucket-here/gazettes/gazete_index.jsonlines` - Where you want the `jsonlines` output for crawls to be saved. This can also be a local location
+    - `SCRAPY_FILES_STORE=s3://name-of-bucket-here/gazettes` - Where you want scraped gazettes to be stored. This can also be a local location
+
+## Running it Locally
+- To run the spider locally, you can choose to store the scraped files locally to do this set the  ENV variable
+- `SCRAPY_FILES_STORE=/directory/to/store/the/files` which should point to a local folder
+- Then run the command  `scrapy crawl tz_gazeyyes_v2 -a year=2016 -o sl_gazettes.jsonlines`          
+   where year is the year you want to scrape gazettes from
+- `sn_gazettes.jsonlines` is the file where crawls are saved, this too can be a directory
```
Comment (DavidLemayian) [inline README.md]: This should match above.

File: opengazettes/settings.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+BOT_NAME = 'opengazettes'
+
+SPIDER_MODULES = ['opengazettes.spiders']
+NEWSPIDER_MODULE = 'opengazettes.spiders'
+FILE_STORE_ACL = 'public-read'
+# DELTAFETCH_ENABLED = True
+
+# Crawl responsibly by identifying yourself (and your website) on the user-agent
+USER_AGENT = 'opengazettes (+http://www.yourdomain.com)'
```
Comment (DavidLemayian) [inline opengazettes/settings.py]: Let's have this as `gazeti.AFRICA (+https://gazeti.africa)`.

**Review by DavidLemayian — CHANGES_REQUESTED** [review]
This is looking good. Just a few slight changes requested.

**Comment by DavidLemayian** [comment]
@ebrahim-j What do you mean "There is a job on scrapinghub to scrape SL gazettes"?

File: README.md (hunk 1 above)
Comment (ebrahim-j) [reply README.md]: Oops, corrected..

File: opengazettes/settings.py (hunk 3 above)
Comment (ebrahim-j) [reply opengazettes/settings.py]: Done

**Comment by ebrahim-j** [comment]
@DavidLemayian The PR has been updated with the **corrected** comment with more detail on it

**Pull request merged**